File: src/my_project/easy_problems/from51to100/count_primes.py
import math

class Solution:
    def countPrimes(self, n: int) -> int:
        if n < 2:
            return 0
        res = [1] * n
        res[0] = 0
        res[1] = 0
        for i in range(2, int(math.sqrt(n)) + 1):
            if res[i] == 1:
                for j in range(i + i, n, i):
                    res[j] = 0
        return sum(res)

File: my_project_typescript/easy_problems_typescript/from1to50/countPrimes.ts
function countPrimes(n: number): number {
  let tempBoolean: boolean = false;

  let count: number = 0;
  // 0, and 1 are not prime numbers

  if (n === 0 || n === 1) {
    return count;
  }

  for (let i = 2; i < n; i++) {
    if (isPrime(i)) {
      count += 1;
    }
  }
  return count;
}

function isPrime(n: number): boolean {
  let isPrimeNumber: boolean = false;

  if (n === 0 || n === 1) {
    return isPrimeNumber;
  }
  if (n === 2) {
    isPrimeNumber = true;
    return isPrimeNumber;
  } 
  
  if (n >= 3) {
    for (let i = 2; i <= Math.sqrt(n); i++) {
      if (n % i === 0) {
        return false;
      }
    }
    isPrimeNumber = true;
    return isPrimeNumber;
  }
  else {
      return false;
  }
}

console.log(countPrimes(10));



